diff --git a/gui/api.py b/gui/api.py
index 7df2b84..b3aa249 100644
--- a/gui/api.py
+++ b/gui/api.py
@@ -3,7 +3,11 @@
 from enum import IntEnum
 from typing import NamedTuple, List, Tuple  # noqa: F401
 
-from _api import *
+try:
+    from _api import *
+except ImportError:
+    # If launched with the replay
+    pass
 
 
 # Vision du Voleur
diff --git a/gui/interface.cc b/gui/interface.cc
index d525522..27a856c 100644
--- a/gui/interface.cc
+++ b/gui/interface.cc
@@ -179,6 +179,8 @@ void api_afficher_position(position v);
 
 /// Affiche le contenu d'une valeur de type perso_info
 void api_afficher_perso_info(perso_info v);
+
+const char* api_get_dump();
 }
 
 template <typename PythonType, typename CxxType>
@@ -1004,6 +1006,13 @@ static PyObject* p_afficher_perso_info(PyObject* /* self */, PyObject* args)
     }
 }
 
+static PyObject* p_get_dump(PyObject* self, PyObject* args)
+{
+    const char* dump = api_get_dump();
+    PyObject* py_dump = PyUnicode_FromString(dump);
+    return py_dump;
+}
+
 // API function to register.
 static PyMethodDef api_callback[] = {
     {"carte_taille", p_carte_taille, METH_VARARGS, "carte_taille"},
@@ -1035,6 +1044,7 @@ static PyMethodDef api_callback[] = {
     {"afficher_erreur", p_afficher_erreur, METH_VARARGS, "afficher_erreur"},
     {"afficher_position", p_afficher_position, METH_VARARGS, "afficher_position"},
     {"afficher_perso_info", p_afficher_perso_info, METH_VARARGS, "afficher_perso_info"},
+    {"get_dump", p_get_dump, METH_VARARGS, "get_dump"},
     {nullptr, nullptr, 0, nullptr}
 };
 
